src/subscript/fmu_copy_revision/fmu_copy_revision.py

```diff
@@ -0,0 +1,654 @@
+    * Directories and files with name 'users'
+    * Directories and files with name 'attic'
+    * Directories and files with names '.git' or '.svn'
+    * Files ending with ~
+    * The following folders under ert/ (if they exist):
+        - output
+        - ert/*/storage, including ert/storage (for backw compat.)
+    * The following folders or files under rms/ (if they exist):
+        - input/seismic, model/*.log
+    * The following files under rms/ (if they exist):
+        - Files under output folders (folders will be kept!)
+    * The following files and folders under spotfire/:
+        - input/*.csv, input/*/.csv model/*.dxp, model/*/*.dxp
+    * The following folders under share/:
+        - results
+        - templates
+    * Empty folders (at destination) except those listed above will kept
+
+4. As profile 3, but all empty folder (at destination) will removed.
+    This the DEFAULT profile!
+
+5. As profile 3, but keeps more data:
+    * Folder rms/output will be copied (~old behaviour)
+    * Folders share/results and share/templates will be copied.
+
+6. Only copy the <coviz> folder (if present), which shall be under
+    <revision>/share/coviz:
+    * Symbolic links will be kept, if possible
+
+9. Make your own filter rules in a named file. For syntax, see e.g.
+    https://www.tutorialspoint.com/unix_commands/rsync.htm
+"""
+
+
+# FILTER* are per file filters, used in the second rsync ommand in the shell file below.
+# DIRFILTER* are per folder, used if DIRTREE == 1 or 2 in the shell script below.
+
+FILTER1 = """
++ **
+"""
+
+FILTER2 = """
+- backup/**
+- users/**
+- attic/**
+- .git/**
+- *.git
+- *.svn
+- *~
+"""
+
+DIRFILTER2 = """
+- backup/
+- users/
+- attic/
+- .git/
++ */
+- *
+"""
+
+FILTER5_ADD = """
++ rms/output/**
++ share/results/**
++ share/templates/**
+"""
+
+FILTER3_ADD = """
+- ert/output/**
+- ert/storage/**
+- ert/output/storage/**
+- input/seismic/**
+- rms/model/*.log
+- rms/output/**
+- spotfire/**/*.csv
+- spotfire/**/*.dxp
+- share/results/**
+- share/templates/**
+"""
+
+DIRFILTER3 = """
+- backup/
+- users/
+- attic/
+- .git/
+- ert/output/
+- ert/storage/
+- ert/**/storage/
+- rms/output
+- rms/input/seismic/
+- share/results/
+- share/templates/
++ */
+- *
+"""
+
+DIRFILTER5 = """
+- backup/
+- users/
+- attic/
+- .git/
+- ert/output/
+- ert/storage/
+- ert/**/storage/
+- rms/input/seismic/
++ */
+- *
+"""
+
+DIRFILTERX = """
++ */
+- *
+"""
+
+FILTER5 = FILTER2 + FILTER3_ADD + FILTER5_ADD
+FILTER3 = FILTER2 + FILTER3_ADD
+
+FILTER6 = """
++ share/coviz/**
+- *
+"""
+
+
+SHELLSCRIPT = """\
+#!/usr/bin/sh
+
+# SETUP OPTIONS
+
+SRCDIR="$1"  # a relative path
+DESTDIR="$2"  # must be an absolute path!
+FILTERFILE="$3"
+THREADS=$4
+RSYNCARGS="$5"
+DIRTREE=$6  # if 1 first copy folder tree, if 2 do it afterwards with dirfilterfile
+DIRFILTERFILE="$7"
+
+PWD=$(pwd)
+
+start=`date +%s.%N`
+
+cd $SRCDIR
+
+echo " ** Target folder is $DESTDIR"
+echo " ** Filter file is $FILTERFILE"
```

Comment:
This filter file is removed if everything works well, so this echo should perhaps be muted (potentially printed as part of a potential error message, to aid debugging)

Reply:
True, it can been seen in the logger if wanted